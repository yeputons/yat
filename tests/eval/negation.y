if (!10) {
    def foo() {
      print 1;
    }
} else {
    def bar() {
      print 0;
    }
}
bar();

if (!0) {
    def foo1() {
       print 2;
    }
} else {
    def bar1() {
       print 3;
    }
}
foo1();

print(!1);
print(!2);
print(5 * !1);
if (!(2 + 2)) { print 10; } else { print 20; }
if (!(2 - 2)) { print 30; } else { print 40; }
print(-(-5));
print(-(-(-(2 * 2))));
