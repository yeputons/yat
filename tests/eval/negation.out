0
2
0
0
0
20
30
5
-4
